### PR TITLE
added plugins section to the manual

### PR DESCRIPTION
**Pull request by abenari**

### CODE REVIEW

File: _includes/manuals/1.4/6_plugins.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+Then restart Foreman with
+<pre>touch ~foreman/tmp/restart.txt</pre>
+
+
+### 6.2 Create a Plugin
+
+This goal of this article is to help developers quickly and easily create a Foreman plugins. This is not an exhaustive tutorial of Ruby on Rails or an explanation of how rails engines work.
+
+As a starting point to build your first Foreman Plugin run the following command
+<pre>rails plugin new my_plugin</pre>
```
Comment (ohadlevy) [inline _includes/manuals/1.4/6_plugins.md]: I would add here that a user needs to install rails first, and maybe we should recommend how to install it (using our packages?)

File: _includes/manuals/1.4/6_plugins.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+A limited number of plugins are available fully packaged from our yum repositories for ease of use. The number of these is increasing, so check the [wiki](http://projects.theforeman.org/projects/foreman/wiki/List_of_Plugins) to see if a package is available yet.  If it's a useful or popular plugin and not yet packaged, please file a feature request in [the packaging project](http://projects.theforeman.org/projects/rpms/issues/new).
+The repos are available at [yum.theforeman.org/plugins](http://yum.theforeman.org/plugins/).  Separate repos are available for each Foreman release, containing plugins that are compatible with that particular version.  Packages are not currently GPG signed.
+
+Configure the repo by creating /etc/yum.repos.d/foreman_plugins.repo:
+
+<pre>
+[foreman-plugins]
+name=Foreman plugins
+baseurl=http://yum.theforeman.org/plugins/1.3/el6/x86_64/
```
Comment (domcleal) [inline _includes/manuals/1.4/6_plugins.md]: Use `{{page.version}}`

File: _includes/manuals/1.4/6_plugins.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+
+Plugins are tools to extend and modify the functionality of Foreman. This article contains a list of plugins by category, and links to other plugin repositories. The core of Foreman is designed to be lean, to maximize flexibility and minimize code bloat. Plugins offer custom functions and features so that each user can tailor their environment to their specific needs. Foreman plugins are implemented as [rail engines](http://guides.rubyonrails.org/engines.html) that are packaged as [gems](http://guides.rubygems.org/what-is-a-gem/) and thus easily installed into Foreman.
```
Comment (domcleal) [inline _includes/manuals/1.4/6_plugins.md]: no list of plugins?

edit: that's not to suggest you should add it, but probably omit this as we don't list them

File: _includes/manuals/1.4/6_plugins.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+### 6.1 Install a Plugin
+
+Foreman plugins are implemented as [gem](http://guides.rubygems.org/what-is-a-gem/) in [Ruby on Rails](http://www.rubyonrails.org). See below for the different installation methods, which depend on your platform.
```
Comment (domcleal) [inline _includes/manuals/1.4/6_plugins.md]: "as gems"

File: _includes/manuals/1.4/6_plugins.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+Not recommended, as it's possible for the 'gem' command to try and install newer versions of Rails which can cause problems. Do note the install without dependencies below to avoid this problem.
+Ensure the plugin you want is available from rubygems.org as a gem. Plugins that aren't published (just git repos) can't be installed with this method without being built as a gem.
+
+If on EL6, run <pre>scl enable ruby193 bash</pre> first for an SCL-enabled shell (not needed on Fedora)
+Install *without* dependencies: <pre>gem install --ignore-dependencies foreman_column_view</pre>
+If you need other dependencies (see the rubygems.org page), check the yum repo above (e.g. deface, nokogiri) or install the same way with 'gem'
+Add to the `bundler.d/Gemfile.local.rb` file as detailed below.
+
+Restart Foreman `touch ~foreman/tmp/restart.txt` or `service httpd restart`
```
Comment (domcleal) [inline _includes/manuals/1.4/6_plugins.md]: "Restart Foreman with `service foreman restart`."

File: _includes/manuals/1.4/6_plugins.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+<pre>gem 'foreman_sample_plugin', :git => "https://github.com/example/foreman_sample_plugin.git"</pre>
+
+Next, as a Foreman user (not root), run the following command:
+<pre>$ bundle update foreman_sample_plugin</pre>
+
+Then restart Foreman with
+<pre>touch ~foreman/tmp/restart.txt</pre>
+
+
+### 6.2 Create a Plugin
```
Comment (domcleal) [inline _includes/manuals/1.4/6_plugins.md]: I don't think this should really be in the manual as it's developer documentation, I'd prefer to see the wiki updated.

File: _includes/manuals/1.4/6_plugins.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+   sub_menu :top_menu, :inner_level, :caption=> N_('Inner level') do
+     menu :top_menu, :level41, :url_hash => {:controller=> :example, :action=>:index}
+     menu :top_menu, :level42, :url_hash => {:controller=> :example, :action=>:index}
+   end
+   menu :top_menu, :level5, :url_hash => {:controller=> :example, :action=>:index}
+ end
+</pre>
+
+### 6.3 Plugin List
+Uptodate plugin list is kept in the [wiki](http://projects.theforeman.org/projects/foreman/wiki/List_of_Plugins)
```
Comment (domcleal) [inline _includes/manuals/1.4/6_plugins.md]: "An up-to-date"

**Comment by domcleal** [comment]
Thanks @abenari!

**Pull request merged**